twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -211,96 +129,193 @@ def _gen_scrooge_srcjar_impl(ctx):
       # In either case (worker or not), they will be jvm flags which will
       # be correctly handled since the executable is a jvm app that will
       # consume the flags on startup.
-      arguments = ["--jvm_flag=%s" % flag for flag in ctx.attr.jvm_flags] +
-      ["@" + argfile.path],
-  )
-
-  deps_jars = collect_jars(ctx.attr.deps)
-
-  scalaattr = struct(
-      outputs = None,
-      compile_jars = deps_jars.compile_jars,
-      transitive_runtime_jars = deps_jars.transitive_runtime_jars,
-  )
-
-  transitive_srcjars = depset(transitive = [
-      collect_srcjars(ctx.attr.deps),
-      collect_extra_srcjars(ctx.attr.deps)
-  ])
-
-  srcjarsattr = struct(
-      srcjar = ctx.outputs.srcjar,
-      transitive_srcjars = transitive_srcjars,
-  )
-
-  return struct(
-      scala = scalaattr,
-      srcjars = srcjarsattr,
-      extra_information = [
-          struct(
-              srcjars = srcjarsattr,
-              scrooge_srcjar = struct(
-                  transitive_owned_srcs = depset(
-                      immediate_thrift_srcs,
-                      transitive = [transitive_owned_srcs])),
-          )
-      ],
+      #arguments = ["--jvm_flag=%s" % flag for flag in ctx.attr.jvm_flags] +
+      arguments = ["@" + argfile.path],
   )
 
-scrooge_scala_srcjar = rule(
-    _gen_scrooge_srcjar_impl,
+def _compiled_jar_file(actions, scrooge_jar):
+  scrooge_jar_name = scrooge_jar.basename
+  # ends with .srcjar, so remove last 6 characters
+  without_suffix = scrooge_jar_name[0:len(scrooge_jar_name) - 6]
+  # this already ends with _scrooge because that is how scrooge_jar is named
+  compiled_jar = without_suffix + "jar"
+  return actions.declare_file(compiled_jar, sibling = scrooge_jar)
+
+def _compile_scala(ctx, label, scrooge_jar, deps_java_info, implicit_deps):
+
+  output = _compiled_jar_file(ctx.actions, scrooge_jar)
+  manifest = ctx.actions.declare_file(
+      label.name + "_MANIFEST.MF", sibling = scrooge_jar)
+  write_manifest_file(ctx.actions, manifest, None)
+  statsfile = ctx.actions.declare_file(
+      label.name + "_scalac.statsfile", sibling = scrooge_jar)
+  merged_deps = java_common.merge(deps_java_info + implicit_deps)
+
+  # We don't need ijar, because generated thrift only changes with the whole thing changes
+  # thrifts generally need the full transitive path to compile
+  compile_scala(
+      ctx,
+      label,
+      output,
+      manifest,
+      statsfile,
+      sources = [],
+      cjars = merged_deps.transitive_compile_time_jars,
+      all_srcjars = depset([scrooge_jar]),
+      buildijar = False,
+      transitive_compile_jars = merged_deps.transitive_compile_time_jars,
+      plugins = [],
+      resource_strip_prefix = "",
+      resources = [],
+      resource_jars = [],
+      labels = {},
+      in_scalacopts = [],
+      print_compile_time = False,
```

Comment:
not for this PR, but with stats files and other things, we should probably just remove this maybe to clean up/fewer options soon

Reply:
I agree. cc @andyscott do we have a ticket to remove this and make sure we can get the data from BEP logs?

Reply:
Is this used for the stats file target? I believe originally this was just a println I added

Reply:
I think this is your println. The statsfile is always emitted I believe.